stubs/2.7/csv.pyi

```diff
@@ -0,0 +1,103 @@
+# Stubs for csv (Python 2.7)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from abc import ABCMeta, abstractmethod
+from typing import Any, Dict, Iterable, List, Sequence, Union
+
+# Public interface of _csv.reader
+class Reader(Iterable[List[str]], metaclass=ABCMeta):
+    dialect = ...  # type: Dialect
+    line_num = ...  # type: int
+
+    @abstractmethod
+    def next(self) -> List[str]: ...
+
+_Row = Sequence[Union[str, int]]
+
+# Public interface of _csv.writer
+class Writer(metaclass=ABCMeta):
+    dialect = ...  # type: Dialect
+
+    @abstractmethod
+    def writerow(self, row: _Row) -> None: ...
+
+    @abstractmethod
+    def writerows(self, rows: List[_Row]) -> None: ...
+
+QUOTE_ALL = ...  # type: int
+QUOTE_MINIMAL = ...  # type: int
+QUOTE_NONE = ...  # type: int
+QUOTE_NONNUMERIC = ...  # type: int
+
+class Error(Exception): ...
+
+_Dialect = Union[str, Dialect]
+
+def writer(csvfile: Any, dialect: _Dialect = ..., **fmtparams) -> Writer: ...
+def reader(csvfile: Iterable[str], dialect: _Dialect = ..., **fmtparams) -> Reader: ...
+def register_dialect(name, dialect=..., **fmtparams): ...
+def unregister_dialect(name): ...
+def get_dialect(name: str) -> Dialect: ...
+def list_dialects(): ...
+def field_size_limit(new_limit: int = ...) -> int: ...
+
+class Dialect:
+    delimiter = ...  # type: str
+    quotechar = ...  # type: str
+    escapechar = ...  # type: str
+    doublequote = ...  # type: bool
+    skipinitialspace = ...  # type: bool
+    lineterminator = ...  # type: str
+    quoting = ...  # type: int
+    def __init__(self) -> None: ...
+
+class excel(Dialect):
+    delimiter = ...  # type: str
```

Comment:
Many of the attributes are inherited; you can just leave them out from base classes so simplify the stub a bit. This also applies to some of the other `Dialect` derived classes.